src/ripple/app/misc/impl/ValidatorList.cpp

```diff
@@ -191,33 +199,159 @@ ValidatorList::load (
     return true;
 }
 
-ListDisposition
+boost::filesystem::path
+ValidatorList::GetCacheFileName(PublicKey const& pubKey)
+{
+    return dataPath_ / (filePrefix_ + strHex(pubKey));
+}
+
+void
+ValidatorList::CacheValidatorFile(PublicKey const& pubKey,
+    PublisherList const& publisher)
+{
+    if (!dataPath_.empty())
+    {
```

Comment:
I'm a fan of early return here, to reduce nesting for the rest of the block and let the reader know there is no (non-trivial) else-branch.